File: Android_Study/MyService/app/src/main/java/com/satelite54/myservice/MainActivity.java
package com.satelite54.myservice;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText editText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        editText = findViewById(R.id.editTextTextPersonName);
        Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = editText.getText().toString();

                Intent intent = new Intent(getApplicationContext(),MyService.class);
                intent.putExtra("command", "show");
                intent.putExtra("name", name);
                startService(intent);
            }
        });
        Intent intent = getIntent();
        onNewIntent(intent);
    }
    // 서비스에서의 인텐트 받기
    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        processIntent(intent);
    }


    public void processIntent(Intent intent) {
        if(intent != null) {
            String command = intent.getStringExtra("command");
            String name = intent.getStringExtra("name");

            Toast.makeText(this, "command : " + command + ", name : " + name, Toast.LENGTH_LONG).show();
        }
    }
}

File: Android_Busan_it/MyService1_Media1/app/src/main/java/com/example/myservice1_media1/MainActivity.kt
package com.example.myservice1_media1

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {

    lateinit var  soundIntent : Intent
    lateinit var btnStart : Button
    lateinit var btnStop : Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        
        title = "서비스 테스트 예제"

        soundIntent = Intent(this, MyService::class.java)
        btnStart = findViewById<Button>(R.id.serviceStart)
        btnStop = findViewById<Button>(R.id.serviceStop)

        btnStart.setOnClickListener {
            startService(soundIntent)
            android.util.Log.i("서비스 테스트", "StartService()")
        }

        btnStop.setOnClickListener {
            stopService(soundIntent)
            android.util.Log.i("서비스 테스트", "StopService()")
        }
    }
}